jsonpb/jsonpb_test.go

```diff
@@ -645,6 +645,29 @@ func TestUnmarshaling(t *testing.T) {
 	}
 }
 
+func TestUnmarshalNullReference(t *testing.T) {
+	var b bytes.Buffer
+
+	for _, ty := range []reflect.Type{
+		reflect.TypeOf(pb.Repeats{}),
+		reflect.TypeOf(pb.Maps{}),
+	} {
+		in := reflect.New(ty).Interface().(proto.Message)
+		if err := (&Marshaler{EmitDefaults: true}).Marshal(&b, in); err != nil {
```

Comment:
I prefer to test unmarshaler by itself and not rely on the Marshaler at all.  Basically, provide JSON strings to unmarshal and compare with expected result.

I also just realized that proto.Equal will not properly compare nil vs empty on these, so I reflect.DeepEqual is ok.